components/org.wso2.carbon.identity.oidc.session/src/test/java/org/wso2/carbon/identity/oidc/session/servlet/OIDCLogoutServletTest.java

```diff
@@ -0,0 +1,242 @@
+
+    @Mock
+    HttpSession httpSession;
+
+    @Mock
+    IdentityConfigParser identityConfigParser;
+
+    @Mock
+    OAuthServerConfiguration oAuthServerConfiguration;
+
+    @Mock
+    KeyStoreManager keyStoreManager;
+
+    @Mock
+    TokenPersistenceProcessor tokenPersistenceProcessor;
+
+    private static final String CLIENT_ID_VALUE = "3T9l2uUf8AzNOfmGS9lPEIsdrR8a";
+    private static final String APP_NAME = "myApp";
+    private static final String SECRET = "87n9a540f544777860e44e75f605d435";
+    private static final String USERNAME = "user1";
+    private static final String OPBROWSER_STATE = "090907ce-eab0-40d2-a46d-acd4bb33f0d0";
+
+    private OIDCLogoutServlet logoutServlet;
+
+    @BeforeTest
+    public void setUp() throws Exception {
+        logoutServlet = new OIDCLogoutServlet();
+
+        initiateInMemoryH2();
+        createOAuthApp(CLIENT_ID_VALUE, SECRET, USERNAME, APP_NAME, "ACTIVE");
+
+    }
+
+    @DataProvider(name = "provideDataForTestDoGet")
+    public Object[][] provideDataForTestDoGet() {
+        Cookie opbsCookie = new Cookie("opbs", OPBROWSER_STATE);
+
+        String idTokenHint =
+                "eyJ4NXQiOiJOVEF4Wm1NeE5ETXlaRGczTVRVMVpHTTBNekV6T0RKaFpXSTRORE5sWkRVMU9HRmtOakZpTVEiLCJr" +
+                        "aWQiOiJOVEF4Wm1NeE5ETXlaRGczTVRVMVpHTTBNekV6T0RKaFpXSTRORE5sWkRVMU9HRmtOakZpTVEiLCJhbGciOiJSUzI1" +
+                        "NiJ9.eyJzdWIiOiJhZG1pbiIsImF1ZCI6WyIzVDlsMnVVZjhBek5PZm1HUzlsUEVJc2RyUjhhIl0sImF6cCI6IjNUOWwydVVmO" +
+                        "EF6Tk9mbUdTOWxQRUlzZHJSOGEiLCJhdXRoX3RpbWUiOjE1MDcwMDk0MDQsImlzcyI6Imh0dHBzOlwvXC9sb2NhbGhvc3Q6OTQ0M" +
+                        "1wvb2F1dGgyXC90b2tlbiIsImV4cCI6MTUwNzAxMzAwNSwibm9uY2UiOiJDcXNVOXdabFFJWUdVQjg2IiwiaWF0IjoxNTA3MDA5ND" +
+                        "A1fQ.ivgnkuW-EFT7m55Mr1pyit1yALwVxrHjVqmgSley1lUhZNAlJMxefs6kjSbGStQg-mqEv0VQ7NJkZu0w1kYYD_76-KkjI1sk" +
+                        "P1zEqSXMhTyE8UtQ-CpR1w8bnTU7D50v-537z8vTf7PnTTA-wxpTuoYmv4ya2z0Rv-gFTM4KPdxsc7j6yFuQcfWg5SyP9lYpJdt-s-O" +
+                        "w9FY1rlUVvNbtF1u2Fruc1kj9jkjSbvFgSONRhizRH6P_25v0LpgNZrOpiLZF92CtkCBbAGQChWACN6RWDpy5Fj2JuQMNcCvkxlv" +
+                        "OVcx-7biH16qVnY9UFs4DxZo2cGzyWbXuH8sDTkzQBg";
+
+        String[] redirectUrl = {
+                "?oauthErrorCode=access_denied&oauthErrorMsg=opbs+cookie+not+received.+Missing+session+state.",
+                "?oauthErrorCode=access_denied&oauthErrorMsg=No+valid+session+found+for+the+received+session+state.",
+                "?oauthErrorCode=server_error&oauthErrorMsg=User+logout+failed",
+                "?oauthErrorCode=access_denied&oauthErrorMsg=End+User+denied+the+logout+request",
+                "https://localhost:8080/playground/oauth2client",
+                "https://localhost:9443/authenticationendpoint/oauth2_logout_consent.do"
+        };
+
+        return new Object[][]{
+                {null, true, redirectUrl[0], "cookie", "", null, false, "", false},
```

Comment:
If possible what each data set corresponds to with regards to the flow. It would be useful when someone else tries to change your test case or even yourself after sometime :wink:

Reply:
+1